ts/features/messages/screens/MessagesSearchScreen.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  Divider,
+  IOSpacingScale,
+  IOStyles,
+  SearchInput,
+  SearchInputRef,
+  VSpacer
+} from "@pagopa/io-app-design-system";
+import React, { useCallback, useRef, useState } from "react";
+import { FlatList, ListRenderItemInfo, Platform, View } from "react-native";
+import { useFocusEffect } from "@react-navigation/native";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+import I18n from "../../../i18n";
+import { EmptyList } from "../components/Search/EmptyList";
+import { useIONavigation } from "../../../navigation/params/AppParamsList";
+import { useIOSelector } from "../../../store/hooks";
+import { searchMessagesCachedSelector } from "../store/reducers/allPaginated";
+import { UIMessage } from "../types";
+import { WrappedMessageListItem } from "../components/Home/WrappedMessageListItem";
+
+const INPUT_PADDING: IOSpacingScale = 16;
+const MIN_QUERY_LENGTH: number = 3;
+
+export const MessagesSearchScreen = () => {
+  const insets = useSafeAreaInsets();
+  const navigation = useIONavigation();
+  const searchInputRef = useRef<SearchInputRef>(null);
+  const [query, setQuery] = useState<string>("");
+
+  const searchResultMessages = useIOSelector(state =>
+    searchMessagesCachedSelector(state, query, MIN_QUERY_LENGTH)
```

Comment:
this selector is rerun on every user input in the searchBar, thus leading  to stuttering and delayed or dropped inputs altogether -- especially when typing fast; I suggest adding a small delay on the selector's execution in order to avoid it, something like:
```ts
 useEffect(() => {
    const delayDebounceFn = setTimeout(() => {
     const searchResult=getSearchResult(query)
     setSearchResult(searchResult)
    }, 250) // I fiddled around a bit, and it feels like the right unnoticeable amount of timeout

    return () => clearTimeout(delayDebounceFn)
  }, [query])
```
this also requires the selector logic to be updated, by either:
- using `store.getState` in the component 
- updating the selector to be a curried function which returns a filtering function, something like:
```ts
// --- selector.ts
const getMessageSearchFn=(state:GlobalState)=>(query:string)=>searchMessage(state,query)
// --- component.ts
const searchFn = useIOSelector(getMessageSearchFn) // then we just call it with the query param

```

Reply:
Nice one, applied